### PR TITLE
Support to specify source paths for unmanaged folder

### LINKED ISSUE #1798: [Feature Request] Expose source paths setting in VS Code for unmanaged folders
Currently we have several classpath related settings for the **unmanaged folders**:
- Default JDK: `java.configuration.runtimes`
- Output path: `java.project.outputPath` (under release)
- Dependencies: `java.project.referencedLibraries`

The missing part is a setting to specify the source paths which originally comes from: https://github.com/microsoft/vscode-java-dependency/issues/412

Though we now have a command `Add/Remove to Java Source Path`, [users are still not aware of this capability](https://stackoverflow.com/q/65140290/13068918). So adding a new setting can better expose such capability.

### PR DESCRIPTION
**Pull request by jdneo**
Resolve #1798 

Related with https://github.com/eclipse/eclipse.jdt.ls/pull/1658

Signed-off-by: Sheng Chen <sheche@microsoft.com>

### CODE REVIEW

File: src/buildpath.ts [hunk 1]
```diff
@@ -23,6 +24,9 @@ export function registerCommands(context: ExtensionContext) {
     context.subscriptions.push(commands.registerCommand(Commands.ADD_TO_SOURCEPATH, async (uri: Uri) => {
         const result = await <any>commands.executeCommand(Commands.EXECUTE_WORKSPACE_COMMAND, Commands.ADD_TO_SOURCEPATH, uri.toString());
         if (result.status) {
+            if (result.sourcePaths) {
```
Comment (jdneo) [inline src/buildpath.ts:27]: The reason for this added patch is that I hope the vscode setting can also be updated if the user changes the source paths from the explorer.

While the side effect is that the setting change event will be synced to the server side and update the classpath again.

**Comment by jdneo** [comment]
👋 @testforstephen @rgrunber Have time to take a look at this PR 😃 ?

File: package.json [hunk 2]
```diff
@@ -359,10 +359,19 @@
             "string",
             "null"
           ],
-          "markdownDescription": "A relative path to the workspace where stores the compiled output. `Only` effective in the `WORKSPACE` scope. The setting will `NOT` affect Maven or Gradle project. ",
+          "markdownDescription": "A relative path to the workspace where stores the compiled output. `Only` effective in the `WORKSPACE` scope. The setting will `NOT` affect Maven or Gradle project.",
           "default": "",
           "scope": "window"
         },
+        "java.project.sourcePaths": {
```
Comment (testforstephen) [inline package.json:366]: Update New settings to README as well.

File: README.md [hunk 3]
```diff
@@ -155,6 +155,8 @@ The following settings are supported:
 * `java.codeGeneration.toString.limitElements`: Limit number of items in arrays/collections/maps to list, if 0 then list all. Defaults to `0`.
 * `java.selectionRange.enabled`: Enable/disable Smart Selection support for Java. Disabling this option will not affect the VS Code built-in word-based and bracket-based smart selection.
 * `java.showBuildStatusOnStart.enabled`: Automatically show build status on startup. Defaults to `false`.
+* `java.project.sourcePaths`: Relative paths to the workspace where stores the source files. `Only` effective in the `WORKSPACE` scope. The setting will `NOT` affect Maven or Gradle project.
```
Comment (testforstephen) [inline README.md:158]: Please refer to https://github.com/redhat-developer/vscode-java/pull/1773/files, usually we put new settings in "**New in 0.77.0**" first.

**Review by testforstephen — APPROVED** [review]
LGTM

**Comment by testforstephen** [comment]
@jdneo could you resolve the conflicts?

**Pull request merged**